shared/js/background/classes/ad-click-attribution-policy.js

```diff
@@ -109,34 +125,62 @@ export class AdClick {
      * @param {number} navigationExpiration in seconds
      * @param {number} totalExpiration in seconds
      */
-    constructor (navigationExpiration, totalExpiration) {
+    constructor (navigationExpiration, totalExpiration, allowlist) {
         /** @type {string | null} */
         this.adBaseDomain = null
         this.adClickRedirect = false
         this.navigationExpiration = navigationExpiration
         this.totalExpiration = totalExpiration
         this.expires = Date.now() + (this.totalExpiration * 1000)
         this.clickExpires = Date.now() + (this.navigationExpiration * 1000)
+        this.allowlist = allowlist
+        this.adClickDNR = null
     }
 
     clone () {
-        const adClick = new AdClick(this.navigationExpiration, this.totalExpiration)
+        const adClick = new AdClick(this.navigationExpiration, this.totalExpiration, this.allowlist)
         adClick.adBaseDomain = this.adBaseDomain
         adClick.adClickRedirect = this.adClickRedirect
         adClick.expires = this.expires
         adClick.clickExpires = Date.now() + (this.navigationExpiration * 1000)
+        adClick.adClickDNR = this.adClickDNR
+        return adClick
+    }
+
+    propagate (tabId) {
+        const adClick = this.clone()
+
+        if (this.adClickDNR) {
+            adClick.adClickDNR = adClick.getAdClickDNR(tabId, this.allowlist)
+            adClick.createDNR()
```

Comment:
I think these calls can be combined.
```suggestion
            adClick.createDNR(tabId, this.allowlist)
```

Reply:
We don't actually need to pass this.allowlist around now we're in the same class.